test/integration/component/test_configdrive.py

```diff
@@ -513,17 +535,22 @@ def update_provider_state(self, new_state):
         :rtype: str
         """
         self.debug("Updating Service Provider ConfigDrive to %s" % new_state)
-        configdriveprovider = NetworkServiceProvider.list(
-            self.api_client,
-            name="ConfigDrive",
-            physicalnetworkid=self.vsp_physical_network.id)[0]
+        configdriveprovider = self.get_configdrive_provider()
         orig_state = configdriveprovider.state
         NetworkServiceProvider.update(self.api_client,
                                       configdriveprovider.id,
                                       state=new_state)
         self.validate_NetworkServiceProvider("ConfigDrive", state=new_state)
         return orig_state
 
+    def _get_test_data(self, key):
+        return self.test_data[key]
+
+    def get_configdrive_provider(self):
+        return NetworkServiceProvider.list(
+            self.api_client,
+            name="ConfigDrive")[0]
+
```

Comment:
Shouldn't it have a line `physicalnetworkid=self.vsp_physical_network.id)[0]`? I am following the same logic of the test `get_configdrive_provider` on `test_nuage_configdrive.py`?

Reply:
Native test can just take the first one, as there are no multiple physical networks which might have a ConfigDrive provider, as is the case in a Nuage setup

Reply:
Got it, thanks.